whitelist/whitelist.go

```diff
@@ -0,0 +1,73 @@
+// Package whitelist provides methods for managing an app's whitelisted IP's.
+package whitelist
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	deis "github.com/deis/controller-sdk-go"
+	"github.com/deis/controller-sdk-go/api"
+)
+
+// List IP's whitelisted for an app.
+func List(c *deis.Client, appID string) (api.Whitelist, error) {
```

Comment:
i have endpoint `/whitelist` which support GET, POST and DELETE in the controller code.Removing this means that i might wanna remove GET in the controller which will not be proper as user had to use `/settings` if he has to get the whitelist.
If i want to not remove the GET from the controller then i might as well use it here since it just returns the whitelist and no the whole appsettings.
The reason for having separate endpoints and not using the existing settings is that if want to delete few addresses in the whitelist then i can't tell using the post request. The other way which we can do it is get the appsettings object delete the addresses in the client side and send a new post request which i don't prefer as logic shouldn't be on the client side.